examples/api-tests/src/scm.spec.js

```diff
@@ -0,0 +1,178 @@
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+// @ts-check
+describe('SCM', function () {
+
+    const { assert } = chai;
+
+    const Uri = require('@theia/core/lib/common/uri');
+    const { ApplicationShell } = require('@theia/core/lib/browser/shell/application-shell');
+    const { ScmContribution } = require('@theia/scm/lib/browser/scm-contribution');
+    const { ScmService } = require('@theia/scm/lib/browser/scm-service');
+    const { ScmWidget } = require('@theia/scm/lib/browser/scm-widget');
+
+    /** @type {import('inversify').Container} */
+    const container = window['theia'].container;
+    const scmContribution = container.get(ScmContribution);
+    const shell = container.get(ApplicationShell);
+
+    /** @type {ScmWidget} */
+    let scmWidget;
+
+    /** @type {ScmService} */
+    let scmService;
+
+    beforeEach(async () => {
+        await shell.leftPanelHandler.collapse();
+        scmWidget = await scmContribution.openView({ activate: true, reveal: true });
+        scmService = scmWidget.getScmService();
+    });
+
+    afterEach(() => {
+        // @ts-ignore
+        scmWidget = undefined;
+        // @ts-ignore
+        scmService = undefined;
+    });
+
+    describe('scm-view', () => {
+
+        it('the view should open and activate successfully', () => {
+            assert.notEqual(scmWidget, undefined);
+            assert.strictEqual(scmWidget, shell.activeWidget);
+        });
+
+        describe('\'ScmTreeWidget\'', () => {
+
+            it('the view should display the resource tree when a repository is present', () => {
+                const scmTreeWidget = scmWidget.getResourceTreeWidget();
+                assert.isTrue(scmTreeWidget.isVisible);
+            });
+
+            it('the view should not display the resource tree when no repository is present', () => {
+
+                // Store the current selected repository so it can be restored.
+                const cachedSelectedRepository = scmService.selectedRepository;
+
+                scmService.selectedRepository = undefined;
+                const scmTreeWidget = scmWidget.getResourceTreeWidget();
+                assert.isFalse(scmTreeWidget.isVisible);
+
+                // Restore the selected repository.
+                scmService.selectedRepository = cachedSelectedRepository;
+            });
+
+        });
+
+        describe('\'ScmNoRepositoryWidget\'', () => {
+
+            it('should not be visible when a repository is present', () => {
+                const noRepositoryWidget = scmWidget['noRepositoryWidget'];
```

Comment:
please check where you access internal APIs and think what should be the proper API, maybe there is already one which you don't use like `ContextKeyService` to evaluate when closures